pkg/command/root.go

```diff
@@ -816,13 +817,29 @@ func shouldSkipGlobalInit(cmd *cobra.Command) bool {
 	return isSkipCommand(skipGlobalInitCommands, cmd.CommandPath())
 }
 
+var globalRootCmd *cobra.Command
+var globalRootCmdLock = &sync.Mutex{}
+
+// NewRootCmd create a new root command instance if it was not created before
+func NewRootCmd() (*cobra.Command, error) {
+	globalRootCmdLock.Lock()
+	defer globalRootCmdLock.Unlock()
+
+	if globalRootCmd == nil {
+		var err error
+		globalRootCmd, err = createRootCmd()
+		return globalRootCmd, err
+	}
+	return globalRootCmd, nil
```

Comment:
Nit: this piece of logic may be cleaner like this:

```
	globalRootCmdLock.Lock()
	defer globalRootCmdLock.Unlock()

	var err error
	if globalRootCmd == nil {
		globalRootCmd, err = createRootCmd()
	}
	return globalRootCmd, err
```
I find it makes it very clear that all the if condition does is set the singleton variable

Reply:
Thanks @marckhouzam for the suggestion. Updated.